lib/skate/secrets_manager.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Skate.SecretsManager do
+  @spec fetch!(String.t()) :: String.t()
+  def fetch!(key_name) do
+    get_secret_value_fn =
+      Application.get_env(:skate, :get_secret_value_fn, &ExAws.SecretsManager.get_secret_value/1)
+
+    aws_request_fn = Application.get_env(:skate, :aws_request_fn, &ExAws.request!/1)
+
+    key_name
+    |> env_specific_key_name()
+    |> get_secret_value_fn.()
+    |> aws_request_fn.()
+    |> Map.fetch!("SecretString")
+  end
+
+  @spec env_specific_key_name(String.t()) :: String.t()
+  def env_specific_key_name(key_name) do
+    env() <> "-" <> secret_key_name(key_name)
+  end
+
+  @spec env() :: String.t()
+  def env do
+    System.get_env("ENVIRONMENT_NAME")
+  end
+
+  @spec secret_key_name(String.t()) :: String.t()
+  def secret_key_name(env_name) do
+    env_name
+    |> String.downcase()
+    |> String.replace("_", "-")
+  end
```

Comment:
Could we put the corrected name in the config instead of having to process it here? So the config would have `{:secret, "api-key"}` instead of `{:secret, "API_KEY"}`?

Reply:
Following up on my comment above, `API_KEY` is actually a good example to show why I kept it this way. The idea was to be consistent between env vars and secret names, which allows you to use different modes per environment (e.g. env vars in dev vs secrets in prod).